phoenix-core/src/main/java/org/apache/phoenix/optimize/QueryOptimizer.java

```diff
@@ -312,6 +320,15 @@ private static QueryPlan addPlan(PhoenixStatement statement, SelectStatement sel
         return null;
     }
 
+    // returns true if we can still use the index
+    // retuns false if we've been in PENDING_DISABLE too long - index should be considered disabled
+    private static boolean isUnderPendingDisableThreshold(PhoenixStatement statement, long currentTimestamp, long indexDisableTimestamp) {
```

Comment:
You could just make addPlan and this method non static and then you could have a member variable for indexPendingDisableThreshold. Might be a little bit cleaner that way IMHO.